aclimatise/converter/janis.py

```diff
@@ -0,0 +1,113 @@
+from typing import List
+
+from aclimatise import cli_types
+from aclimatise.cli_types import CliType
+from aclimatise.converter import NamedArgument, WrapperGenerator
+from aclimatise.model import CliArgument, Command, Flag, Positional
+
+
+class JanisGenerator(WrapperGenerator):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        import janis_core as j
+
+        # save importing janis until this is initialised
+        self.janis = j
+
+    @classmethod
+    def format(cls) -> str:
+        return "janis"
+
+    def save_to_string(self, cmd: Command) -> str:
+
+        clt = self.command_to_tool(cmd)
+        return clt.translate("janis", to_console=False)
+
+    def command_to_tool(self, cmd: Command):
+
+        inputs: List[CliArgument] = [*cmd.named] + (
+            [] if self.ignore_positionals else [*cmd.positional]
+        )
+        names = self.choose_variable_names(inputs)
+
+        tool = self.janis.CommandToolBuilder(
+            tool=cmd.as_filename,
+            base_command=list(cmd.command),
+            inputs=self.get_inputs(names),
+            outputs=self.get_outputs(names),
+            version="v0.1.0",
+            container=cmd.docker_image,
+        )
+
+        return tool
+
+    def type_to_cwl_type(self, typ: cli_types.CliType, optional):
```

Comment:
Is this function misnamed? Should it be janis type?

Reply:
Also would appreciate a type annotation on the args and output here

Reply:
It should be `janis_type` and I've added type annotations.